src/corehost/common/longfile.windows.cpp

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+//The logic in this file was ported from https://github.com/dotnet/coreclr/blob/54891e0650e69f08832f75a40dc102efc6115d38/src/utilcode/longfilepathwrappers.cpp
```

Comment:
I wanted to capture the point in time when we actually ported the logic.
Also this ensures that if the file is renamed or moved, we can always find the original source